js/tests/mega-menu/tests.js

```diff
@@ -0,0 +1,253 @@
+QUnit.module("The HTML structure is as expected.");
+
+QUnit.test("The nav element is present", function (assert) {
+
+    assert.equal($('nav').attr('id'), 'nav', "The nav has an attribute of id which is #nav.");
+
+});
+
+QUnit.test("Tests for necessary number of children", function (assert) {
+
+    $('.mega-menu > ul').children().each(function () {
+
+        var fixtureLength = $(this).children().length;
+
+        assert.ok(fixtureLength == 2, "The '.mega-menu > ul' has 2 fixtures inside");
+
+    })
+});
+
+QUnit.test("Tests for necessary children name", function (assert) {
+
+    $('.mega-menu > ul').children().each(function () {
+
+        var firstNodeName = $(this).children().first()[0].nodeName;
+
+        var lastNodeName = $(this).children().last()[0].nodeName;
+
+        assert.equal(firstNodeName, 'A', 'This gets the first child which should be A');
```

Comment:
node.nodeName will always return the element node in uppercase.
It will fail if the comparison value is in lowercase, as JS is case-sensitive.